File: midolman/src/main/java/org/midonet/midolman/rules/ChainPacketContext.java
// Copyright 2012 Midokura Inc.

package org.midonet.midolman.rules;

import java.util.Set;
import java.util.UUID;

import org.midonet.util.functors.Callback0;

public interface ChainPacketContext {
    UUID getInPortId();
    UUID getOutPortId();
    Set<UUID> getPortGroups();
    void addTraversedElementID(UUID id);
    void addFlowRemovedCallback(Callback0 cb);
    void addFlowTag(Object tag);
    boolean isConnTracked();
    boolean isForwardFlow();
    Integer getFlowCookie();
    Integer getParentCookie();
}

File: midolman/src/main/scala/org/midonet/midolman/rules/ChainPacketContext.scala
// Copyright 2013 Midokura Inc.

package org.midonet.midolman.rules

import java.util.{Set => JSet}
import java.util.UUID

import org.midonet.util.functors.Callback0
import org.midonet.sdn.flows.FlowTagger
import FlowTagger.FlowTag
;

trait ChainPacketContext {
    def inPortId: UUID
    def outPortId: UUID
    def portGroups: JSet[UUID]
    def isConnTracked: Boolean
    def isForwardFlow: Boolean
    def flowCookie: Option[Int]
    def parentCookie: Option[Int]

    def addFlowTag(tag: FlowTag)
    def addFlowRemovedCallback(cb: Callback0)
}
